do not assign pure function components a ref

pwmckenna opened the pull request:
happy to adjust line length stuff, or c+p those tests into each package that does this...or anything else really. :evergreen_tree: 

fixes https://github.com/yahoo/fluxible/issues/377

yahoocla commented:
Thank you for submitting this pull request, however I do not see a valid CLA on file for you. Before we can merge this request please visit https://yahoocla.herokuapp.com/ and agree to the terms. Thanks! :smile:

yahoocla commented:
CLA is valid!

mridgway commented on packages/fluxible-addons-react/connectToStores.js:
```diff
@@ -53,7 +53,7 @@ function createComponent(Component, stores, getStateFromStores, customContextTyp
         },
         render: function render() {
-            return React.createElement(Component, Object.assign({}, this.props, this.state, {ref: 'wrappedElement'}));
+            return React.createElement(Component, Object.assign({}, this.props, this.state, Component.prototype && Component.prototype.isReactComponent ? {ref: 'wrappedElement'} : null));
```
Let's pull this out into a separate variable:

`var props = Component.prototype && Component.prototype.isReactComponent ? {ref: 'wrappedElement'} : null;`

gingur replied:
+1 for breaking logic into separate variable.

mridgway commented:
Looks good. We should pull the props out to a separate variable. And yeah, we should have a test for each component if you don't mind.

pwmckenna commented:
updated with seperate `props` variables, and added tests for each usage. There was some variation between how the packages tested, so I tried to conform to each of the packages' styles.

mridgway commented:
:+1: awesome work @pwmckenna! Will wait for tests to pass and get this merged.
